samples/subsys/fs/littlefs/src/main.c

```diff
@@ -31,132 +270,104 @@ static struct fs_mount_t lfs_storage_mnt = {
+		LOG_PRINTK("FAIL: mount id %" PRIuPTR " at %s: %d\n",
 		       (uintptr_t)mp->storage_dev, mp->mnt_point, rc);
-		return;
+		return rc;
 	}
-	printk("%s mount: %d\n", mp->mnt_point, rc);
+	LOG_PRINTK("%s mount: %d\n", mp->mnt_point, rc);
 #else
-	printk("%s automounted\n", mp->mnt_point);
+	LOG_PRINTK("%s automounted\n", mp->mnt_point);
 #endif
 
-	rc = fs_statvfs(mp->mnt_point, &sbuf);
-	if (rc < 0) {
-		printk("FAIL: statvfs: %d\n", rc);
-		goto out;
-	}
+	return 0;
+}
+#endif /* CONFIG_APP_LITTLEFS_STORAGE_FLASH */
 
-	printk("%s: bsize = %lu ; frsize = %lu ;"
-	       " blocks = %lu ; bfree = %lu\n",
-	       mp->mnt_point,
-	       sbuf.f_bsize, sbuf.f_frsize,
-	       sbuf.f_blocks, sbuf.f_bfree);
+#ifdef CONFIG_APP_LITTLEFS_STORAGE_BLK_SDMMC
+struct fs_littlefs lfsfs;
+static struct fs_mount_t __mp = {
+	.type = FS_LITTLEFS,
+	.fs_data = &lfsfs,
+	.flags = FS_MOUNT_FLAG_USE_DISK_ACCESS,
+};
+struct fs_mount_t *mp = &__mp;
 
-	struct fs_dirent dirent;
+static int littlefs_mount(struct fs_mount_t *mp)
+{
+	static const char *disk_mount_pt = "/"CONFIG_SDMMC_VOLUME_NAME":";
+	static const char *disk_pdrv = CONFIG_SDMMC_VOLUME_NAME;
 
-	rc = fs_stat(fname, &dirent);
-	printk("%s stat: %d\n", fname, rc);
-	if (rc >= 0) {
-		printk("\tfn '%s' size %zu\n", dirent.name, dirent.size);
-	}
+	mp->storage_dev = (void *) disk_pdrv;
+	mp->mnt_point = disk_mount_pt;
 
-	struct fs_file_t file;
+	return fs_mount(mp);
+}
+#endif /* CONFIG_APP_LITTLEFS_STORAGE_BLK_SDMMC */
 
-	fs_file_t_init(&file);
+void main(void)
+{
+	char fname1[MAX_PATH_LEN];
+	char fname2[MAX_PATH_LEN];
+	struct fs_statvfs sbuf;
+	int rc;
 
-	rc = fs_open(&file, fname, FS_O_CREATE | FS_O_RDWR);
+	LOG_PRINTK("Sample program to r/w files on littlefs\n");
+
+	rc = littlefs_mount(mp);
 	if (rc < 0) {
-		printk("FAIL: open %s: %d\n", fname, rc);
-		goto out;
+		return;
 	}
 
-	uint32_t boot_count = 0;
-
-	if (rc >= 0) {
-		rc = fs_read(&file, &boot_count, sizeof(boot_count));
-		printk("%s read count %u: %d\n", fname, boot_count, rc);
-		rc = fs_seek(&file, 0, FS_SEEK_SET);
-		printk("%s seek start: %d\n", fname, rc);
+	snprintf(fname1, sizeof(fname1), "%s/boot_count", mp->mnt_point);
+	snprintf(fname2, sizeof(fname2), "%s/pattern.bin", mp->mnt_point);
 
+	rc = fs_statvfs(mp->mnt_point, &sbuf);
+	if (rc < 0) {
+		LOG_PRINTK("FAIL: statvfs: %d\n", rc);
+		goto out;
 	}
 
-	boot_count += 1;
-	rc = fs_write(&file, &boot_count, sizeof(boot_count));
-	printk("%s write new boot count %u: %d\n", fname,
-	       boot_count, rc);
-
-	rc = fs_close(&file);
-	printk("%s close: %d\n", fname, rc);
-
-	struct fs_dir_t dir;
-
-	fs_dir_t_init(&dir);
-
-	rc = fs_opendir(&dir, mp->mnt_point);
-	printk("%s opendir: %d\n", mp->mnt_point, rc);
+	LOG_PRINTK("%s: bsize = %lu ; frsize = %lu ;"
+		   " blocks = %lu ; bfree = %lu\n",
+		   mp->mnt_point,
+		   sbuf.f_bsize, sbuf.f_frsize,
+		   sbuf.f_blocks, sbuf.f_bfree);
 
-	while (rc >= 0) {
-		struct fs_dirent ent = { 0 };
+	lsdir(mp->mnt_point);
```

Comment:
You are not checking result of the function here so you want to proceed whether this failed or not?